File: src/main/java/com/hohltier/decompiler/controllers/WindowController.java
package com.hohltier.decompiler.controllers;

import com.hohltier.decompiler.views.WindowView;
import lombok.Getter;

public class WindowController extends BaseController<WindowView> {

    @Getter private static final WindowController INSTANCE = new WindowController();

    private WindowController() {
        super(new WindowView());
    }

    public void show() {
        getView().setVisible(true);
    }

    public void activate() {
        getView().setContentPane(getView().getSplitPane());
        getView().validate();
    }

    public void dispose() {
        getView().dispose();
    }

}

File: src/main/kotlin/moe/sota/decompiler/controllers/WindowController.kt
package moe.sota.decompiler.controllers

import moe.sota.decompiler.views.WindowView

class WindowController(
    private val windowView: WindowView
) {

    fun show() {
        windowView.isVisible = true
    }

    fun activate() {
        windowView.contentPane = windowView.splitPane
        windowView.validate()
    }

    fun dispose() {
        windowView.dispose()
    }

}
